src/main/java/com/juniorstart/juniorstart/controller/UserProfileController.java

```diff
@@ -24,4 +33,29 @@ public UserProfileController(UserProfileService userProfileService) {
 
         return userProfileService.selectionForSearching(userRoleOrTechnologyRequest, page, size);
     }
+
+    @GetMapping("/avatar")
+    public ResponseEntity<?> getUserAvatar(@CurrentUser UserPrincipal userPrincipal) {
+
+        return userProfileAvatarService.getUserAvatar(userPrincipal);
```

Comment:
nie dajesz / w Get Post itp bo spring automatycznie sam dodaje. Poza tym jak wszędzie używasz avatar to wystarczy dać adnotacje nad klasą @RequestMapping("avatar ") i domyślnie będzie wszędzie taka sama.

Reply:
Tak, tylko tam jeszcze jest inna metoda(gryzły by się), która jest do refaktoryzacji.

Reply:
która?

Reply:
`    @PreAuthorize("hasRole('USER')")
    @GetMapping("/technology_role")
    public PagedResponse<UserProfile.UserProfileDto> findByUserRolesAndTechnologies(@RequestBody UserRoleOrTechnologyRequest userRoleOrTechnologyRequest,
                                                                                    @RequestParam(value = "page", defaultValue = AppConstants.DEFAULT_PAGE_NUMBER) int page,
                                                                                    @RequestParam(value = "size", defaultValue = AppConstants.DEFAULT_PAGE_SIZE) int size) {

        return userProfileService.selectionForSearching(userRoleOrTechnologyRequest, page, size);
    }`